### PR TITLE
Feature: debug shell

### PR DESCRIPTION
**Pull request by kvaps**
**Description**

This PR adds two new options: `--debug` and `--debug-shell` which allow to run interactive shell in case of RUN command failed.
This option might be very useful during the development the Dockerfile. I'm going to implement it's support into [`kubectl-build`](https://github.com/kvaps/kubectl-build) plugin.

Live demo:
[asciicast](https://asciinema.org/a/432020)

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

**Release Notes**

Describe any changes here so maintainer can include it in the release notes, or delete this block.

```
Examples of user facing changes:
- kaniko adds a new flag `--debug` and `--debug-cmd` to spawn interactive debug shell in case of RUN command failed
```

### CODE REVIEW

File: pkg/executor/build.go [hunk 1]
```diff
@@ -370,6 +373,12 @@ func (s *stageBuilder) build() error {
 		}
 
 		if err := command.ExecuteCommand(&s.cf.Config, s.args); err != nil {
+			if s.opts.Debug {
+				logrus.Errorf(errors.Wrap(err, "Failed to execute command").Error())
+				if err := startDebugShell(s.opts.DebugCmd); err != nil {
+					return errors.Wrap(err, "error starting debug shell")
+				}
+			}
```
Comment (kvaps) [inline pkg/executor/build.go:381]: I'm thinking about moving `startDebugShell()` under `command.ExecuteCommand()` and reuse `runCommandInExec()` which is actually loading a correct context with all the ENVs and ARGs set.

https://github.com/GoogleContainerTools/kaniko/blob/da50668f1eaffd16149719e275ba8d084b8d3e83/pkg/commands/run.go#L48-L50

But there is no options to pass command line arguments. Or I can modify `runCommandInExec()` to move all the common logic to separate function under `util` package.

What is the best option for this?

**Comment by kvaps** [comment]
PR rebased to `v1.7.0`, new docker images with this change prepared:

```
ghcr.io/kvaps/kaniko-executor:v1.7.0
ghcr.io/kvaps/kaniko-executor:v1.7.0-debug
ghcr.io/kvaps/kaniko-warmer:v1.7.0
```

source code of fork in `master` branch:
https://github.com/kvaps/kaniko/

**Comment by tejal29** [comment]
rebasing with latest master

**Comment by gabyx** [comment]
Awespme!

**Comment by aaron-prindle** [comment]
@kvaps would you be able to rebase the PR here? Would like to get this in if possible, thanks!

**Pull request closed without merging**